math/src/field/test_fields/u64_test_field.rs

```diff
@@ -0,0 +1,59 @@
+use crate::field::traits::{IsField, TwoAdicField};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct U64TestField<const MODULUS: u64>;
+
+impl<const MODULUS: u64> IsField for U64TestField<MODULUS> {
+    type BaseType = u64;
+
+    fn add(a: &u64, b: &u64) -> u64 {
+        ((*a as u128 + *b as u128) % MODULUS as u128) as u64
+    }
+
+    fn sub(a: &u64, b: &u64) -> u64 {
+        (((*a as u128 + MODULUS as u128) - *b as u128) % MODULUS as u128) as u64
+    }
+
+    fn neg(a: &u64) -> u64 {
+        MODULUS - a
+    }
+
+    fn mul(a: &u64, b: &u64) -> u64 {
+        ((*a as u128 * *b as u128) % MODULUS as u128) as u64
+    }
+
+    fn div(a: &u64, b: &u64) -> u64 {
+        Self::mul(a, &Self::inv(b))
+    }
+
+    fn inv(a: &u64) -> u64 {
+        assert_ne!(*a, 0, "Cannot invert zero element");
```

Comment:
Do we have a safe version of this? Providing only panicking options is frowned upon for libraries. You should always give the caller an option to catch mistakes.

Reply:
Yeah we should return an error here instead, while we're at it we should do the same for `U64PrimeField` and `MontgomeryBackendPrimeField`

Reply:
Let's do this on a separate PR to fix them all

Reply:
This is now contemplated in issue #77, it can't be fixed here because it would require to change the `IsField` trait, and consequently every field implementation.